spring-kafka/src/main/java/org/springframework/kafka/config/KafkaBackOffManagerConfigurationSupport.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.kafka.listener.KafkaConsumerTimingAdjuster;
+import org.springframework.kafka.listener.ListenerContainerRegistry;
+import org.springframework.kafka.listener.PartitionPausingBackoffManager;
+import org.springframework.kafka.listener.WakingKafkaConsumerTimingAdjuster;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.util.Assert;
+
+/**
+ * This class provides configuration support for a global
+ * {@link KafkaConsumerBackoffManager} instance. Consider overriding any of the
+ * protected methods for providing different components or configuration.
+ * This class is automatically imported by the
+ * {@link org.springframework.kafka.annotation.EnableRetryTopic @EnableRetryTopic}
+ * annotation.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.9
+ */
+public class KafkaBackOffManagerConfigurationSupport implements ApplicationContextAware {
+
+	private ConfigurableApplicationContext applicationContext;
+
+	/**
+	 * Provides the {@link KafkaConsumerBackoffManager} instance.
+	 * To customize it or any of the components, consider overriding
+	 * one of the more fine graned methods:
+	 * <ul>
+	 *     <li>{@link #backOffManagerClock}</li>
+	 *     <li>{@link #timingAdjuster}</li>
+	 *     <li>{@link #timingAdjusterTaskExecutor}</li>
+	 * </ul>
+	 * @param registry the global {@link ListenerContainerRegistry} instance.
+	 * @return the instance.
+	 */
+	@Bean(name = KafkaListenerConfigUtils.KAFKA_CONSUMER_BACK_OFF_MANAGER_BEAN_NAME)
+	public KafkaConsumerBackoffManager kafkaConsumerBackoffManager(@Qualifier(KafkaListenerConfigUtils.KAFKA_LISTENER_ENDPOINT_REGISTRY_BEAN_NAME)
+																ListenerContainerRegistry registry) {
+		return new PartitionPausingBackoffManager(registry, timingAdjuster(timingAdjusterTaskExecutor()), backOffManagerClock());
+	}
+
+	/**
+	 * Override this method to provide a different {@link Clock}
+	 * instance to be used with the {@link KafkaConsumerBackoffManager}.
+	 * @return the instance.
+	 */
+	protected Clock backOffManagerClock() {
+		return Clock.systemUTC();
+	}
+
+	/**
+	 * Override this method to provide a different {@link KafkaConsumerTimingAdjuster}
+	 * to be used with the {@link KafkaConsumerBackoffManager}.
+	 * @param taskExecutor the task executor.
+	 * @return the instance.
+	 */
+	protected KafkaConsumerTimingAdjuster timingAdjuster(TaskExecutor taskExecutor) {
+		return new WakingKafkaConsumerTimingAdjuster(taskExecutor);
+	}
+
+	/**
+	 * Override this method to provide a different {@link TaskExecutor}
+	 * to be used with the {@link KafkaConsumerTimingAdjuster}.
+	 * @return the instance.
+	 */
+	protected TaskExecutor timingAdjusterTaskExecutor() {
+		ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+		executor.initialize();
+		this.applicationContext
+				.addApplicationListener((ApplicationListener<ContextClosedEvent>) event -> executor.shutdown());
```

Comment:
This is a bit awkward.
Consider to implement `DisposableBean` instead.

Reply:
I agree this looks awkward, just couldn't think of a better way 😄 You mean implementing `DisposableBean` in the `Support` class and keeping a reference to the `TaskExecutor` to `destroy()` it later?

Reply:
Right. Why not?

Reply:
See the same `WebMvcConfigurationSupport`. It does have a state via its properties:
```
	@Nullable
	private List<Object> interceptors;

	@Nullable
	private PathMatchConfigurer pathMatchConfigurer;

	@Nullable
	private ContentNegotiationManager contentNegotiationManager;
```

Reply:
Oh, sure, makes sense. I'll change this, thanks.